### PR TITLE
Added AVL Tree in C

### PR DESCRIPTION
**Pull request by yashsahay25**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added implementation of AVL Tree.
AVL tree is a height-balanced binary search tree (BST) where the Balance Factor ( the difference between heights of left and right subtrees) cannot be more than one for all nodes.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

File: C/graphs/AVL_Tree.c [hunk 1]
```diff
@@ -0,0 +1,405 @@
+/*Progrm for AVL Tree*/
```
Comment (iamrajiv) [inline C/graphs/AVL_Tree.c]: @yashsahay2014 fix `Progrm` typo

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@yashsahay2014 Add sample input-output.

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@yashsahay2014 Please do following modifications
* Add the Pre & Post Order traversal of the Constructed AVL Tree
* Add more suitable comments in the Code

**Comment by yashsahay25** [comment]
@iamrajiv @siddharth25pandey Sorry for the delay. I have made the requested changes. Kindly review them.

File: C/graphs/AVL_Tree.c [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+	
+	Constructed AVL tree would be
+		 48
+		/  \
+	      33    58
+	     /  \   / \
+	    30  40 50 60
+	   /    / \
+	  25   35  42
+------------------------------------------------*/
```
Comment (iamrajiv) [inline C/graphs/AVL_Tree.c]: @yashsahay2014 remove this type of line pattern looks unprofessional

File: C/graphs/AVL_Tree.c [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+	    33  40 50 
+	        
+	root = del(root,35); 
+	root = del(root,33); 
+	root = del(root,58); 
+	Modified AVL tree would be
+		 42
+		/  \
+	      40    50
+------------------------------------------------*/
```
Comment (iamrajiv) [inline C/graphs/AVL_Tree.c]: remove this too

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@yashsahay2014 Add space and time complexity like time for rotate operations, updating the height, AVL insert etc.

**Comment by yashsahay25** [comment]
@iamrajiv I have made the requested changes and also included Time & Space complexity. But there is Merge Conflicts Labeler error. Kindly help me & how to solve that error?

**Comment by iamrajiv** [comment]
@yashsahay2014 please update the readme

**Comment by yashsahay2014** [comment]
@iamrajiv Done

**Comment by iamrajiv** [comment]
@yashsahay2014 Update this https://github.com/TesseractCoding/NeoAlgo/blob/master/C/README.md

**Comment by yashsahay25** [comment]
@iamrajiv Do I need to make a new PR for that? Kindly guide me.

**Comment by iamrajiv** [comment]
@yashsahay2014 no just go to your branch and update the `C/README.md` file. Updating `C/README.md` file means add AVL Tree program link in graphs section.

**Comment by iamrajiv** [comment]
@yashsahay2014 I have updated the readme

**Comment by yashsahay25** [comment]
> @yashsahay2014 no just go to your branch and update the `C/README.md` file. Updating `C/README.md` file means add AVL Tree program link in graphs section.

@iamrajiv I have also done this. Thanks for guiding me. Kindly merge it into master.

**Review by iamrajiv — APPROVED** [review]

**Comment by iamrajiv** [comment]
@yashsahay2014 I have approved the changes. Wait for others to review the changes.

**Comment by yashsahay25** [comment]
@siddharth25pandey @SKAUL05 kindly review the PR

**Pull request closed without merging**